src/xdp-utils.c

```diff
@@ -2151,7 +2151,22 @@ app_info_map_pids (XdpAppInfo  *app_info,
   g_return_val_if_fail (app_info != NULL, FALSE);
   g_return_val_if_fail (pids != NULL, FALSE);
 
-  if (app_info->kind != XDP_APP_INFO_KIND_FLATPAK)
+  if (app_info->kind == XDP_APP_INFO_KIND_SNAP)
+    {
+      for (int i = 0; i < n_pids; i++)
+        {
+          XdpAppInfo *info = xdp_get_app_info_from_pid (pids[i], error);
```

Comment:
You're potentially overwriting `error` here, you need to check if info is valid too.

Reply:
This still stands.

If `xdp_get_app_info_from_pid` fails, you'll have the `error` set, thus the next `g_set_error_literal` will fail as you're overwriting it.

So I think it would be better to propagate it instead.

Reply:
Fixed in https://github.com/flatpak/xdg-desktop-portal/commit/3055e448bd61dd551be6fce48d15ce70751182f6